src/dist/component/package.rs

```diff
@@ -384,16 +386,20 @@ fn unpack_without_first_dir<'a, R: Read>(
                 trigger_children(&*io_executor, &mut directories, &mut budget, op)?;
             }
             // Maybe stream a file incrementally
-            if let Some((sender, entry)) = sender_entry.as_mut() {
+            if let Some(sender) = sender_entry.as_mut() {
                 if budget.available() as u64 >= IO_CHUNK_SIZE {
                     let mut v = vec![0; IO_CHUNK_SIZE as usize];
```

Comment:
If we're using `read_to_end` then there's little point in initialising the vec here. The `.take()` will hint the size to `read_to_end()` and we won't have to waste zero-initialising the buffer.  This may as well be `Vec::new()`
```suggestion
                    let mut v = Vec::new();
```

Reply:
It would be great if it did. But nopity nope nope nope nope. https://doc.rust-lang.org/src/std/io/mod.rs.html#355

Reply:
https://doc.rust-lang.org/src/std/io/mod.rs.html#2416 sorry ... but still, note it takes the min. Not the max.